gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageFileSystemProvider.java

```diff
@@ -108,7 +94,21 @@ public CloudStorageFileSystem getFileSystem(URI uri) {
   }
 
   /**
-   * Returns Cloud Storage file system, provided a URI with no path, e.g. {@code gs://bucket}.
+   * Returns Cloud Storage file system, provided a URI with no path.
+   *
+   * <p><b>Note:</b> This method should be invoked indirectly via the SPI by calling
```

Comment:
Technically `FileSystems.newFileSystem` is an API, not the SPI. The SPI is stuff like `FileSystemProvider` that users shouldn't ever interact with directly.